File: src/main/java/net/atos/client/bag/util/StatusVerblijfsobjectEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.StatusVerblijfsobject;

public class StatusVerblijfsobjectEnumAdapter implements JsonbAdapter<StatusVerblijfsobject, String> {

    @Override
    public String adaptToJson(final StatusVerblijfsobject statusVerblijfsobject) {
        return statusVerblijfsobject.toString();
    }

    @Override
    public StatusVerblijfsobject adaptFromJson(final String json) {
        return StatusVerblijfsobject.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/StatusVerblijfsobjectEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.StatusVerblijfsobject

class StatusVerblijfsobjectEnumAdapter : JsonbAdapter<StatusVerblijfsobject, String> {
    override fun adaptToJson(statusVerblijfsobject: StatusVerblijfsobject): String = statusVerblijfsobject.toString()
    override fun adaptFromJson(json: String): StatusVerblijfsobject = StatusVerblijfsobject.fromValue(json)
}
